chia/cmds/wallet_funcs.py

```diff
@@ -528,32 +530,34 @@ async def make_offer(
                         converted_amount = Decimal(requested_amount) / divisor
                         print(f"  - {converted_amount} {asset} ({requested_amount} mojos)")
 
-                    print()
-                    nft_confirmation = input(
-                        "Offers for NFTs will have royalties automatically added. "
-                        + "Are you sure you would like to continue? (y/n): "
-                    )
-                    if nft_confirmation not in ["y", "yes"]:
+                    if not no_confirm:
+                        print()
+                        nft_confirmation = input(
+                            "Offers for NFTs will have royalties automatically added. "
+                            + "Are you sure you would like to continue? (y/n): "
+                        )
+                        if nft_confirmation not in ["y", "yes"]:
+                            print("Not creating offer...")
+                            return
+
+                if not no_confirm:
+                    confirmation = input("Confirm (y/n): ")
+                    if confirmation not in ["y", "yes"]:
                         print("Not creating offer...")
                         return
-
-                confirmation = input("Confirm (y/n): ")
-                if confirmation not in ["y", "yes"]:
-                    print("Not creating offer...")
-                else:
-                    with open(pathlib.Path(filepath), "w") as file:
-                        offer, trade_record = await wallet_client.create_offer_for_ids(
-                            offer_dict, driver_dict=driver_dict, fee=fee, reuse_puzhash=reuse_puzhash
-                        )
-                        if offer is not None:
+                offer, trade_record = await wallet_client.create_offer_for_ids(
+                    offer_dict, driver_dict=driver_dict, fee=fee, reuse_puzhash=reuse_puzhash
+                )
+                if offer is not None:
+                    if not no_file:
```

Comment:
is the `--no-file` feature important for the tests?
I would think you could just ask for a temporary directory and write the file there

Reply:
yeah i could do that.

Reply:
oh wait i remember why, by using the --no-file argument i would not have to fake the part that compresses the offer and turns it into bytes.

Reply:
unless it would be insurmountable to fake, I think that would be a better approach. You wouldn't have to add test code mixed in to the production code (and you would get better test coverage metric).

I would expect that you don't actually need to save the offer in any correct format, or anything but just an empty buffer perhaps

Reply:
yeah i went back with a fresh mind and it was easy, fixed,